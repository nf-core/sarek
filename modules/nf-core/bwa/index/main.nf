process BWA_INDEX {
    tag "$fasta"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/bwa:0.7.18--he4a0461_0' :
        'biocontainers/bwa:0.7.18--he4a0461_0' }"

    input:
    tuple val(meta), path(fasta)

    output:
    tuple val(meta), path(bwa) , emit: index
    path "versions.yml"        , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def prefix = task.ext.prefix ?: "${fasta.baseName}"
    def args   = task.ext.args ?: ''
    """
    mkdir bwa
    bwa \\
        index \\
        $args \\
        -p bwa/${prefix} \\
        $fasta

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        bwa: \$(echo \$(bwa 2>&1) | sed 's/^.*Version: //; s/Contact:.*\$//')
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${fasta.baseName}"
    """
    mkdir bwa

    touch bwa/${prefix}.amb
    touch bwa/${prefix}.ann
    touch bwa/${prefix}.bwt
    touch bwa/${prefix}.pac
    touch bwa/${prefix}.sa

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        bwa: \$(echo \$(bwa 2>&1) | sed 's/^.*Version: //; s/Contact:.*\$//')
    END_VERSIONS
    """
}
