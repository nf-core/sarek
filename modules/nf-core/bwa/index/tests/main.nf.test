nextflow_process {

    name "Test Process BWA_INDEX"
    tag "modules_nfcore"
    tag "modules"
    tag "bwa"
    tag "bwa/index"
    script "../main.nf"
    process "BWA_INDEX"

    test("BWA index") {

        when {
            process {
                """
                input[0] = [
                    [id: 'test'],
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
