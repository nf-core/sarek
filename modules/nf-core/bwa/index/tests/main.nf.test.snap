{
    "BWA index": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "genome.amb:md5,3a68b8b2287e07dd3f5f95f4344ba76e",
                            "genome.ann:md5,c32e11f6c859f166c7525a9c1d583567",
                            "genome.bwt:md5,0469c30a1e239dd08f68afe66fde99da",
                            "genome.pac:md5,983e3d2cd6f36e2546e6d25a0da78d66",
                            "genome.sa:md5,ab3952cabf026b48cd3eb5bccbb636d1"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,a64462ac7dfb21f4ade9b02e7f65c5bb"
                ],
                "index": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "genome.amb:md5,3a68b8b2287e07dd3f5f95f4344ba76e",
                            "genome.ann:md5,c32e11f6c859f166c7525a9c1d583567",
                            "genome.bwt:md5,0469c30a1e239dd08f68afe66fde99da",
                            "genome.pac:md5,983e3d2cd6f36e2546e6d25a0da78d66",
                            "genome.sa:md5,ab3952cabf026b48cd3eb5bccbb636d1"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,a64462ac7dfb21f4ade9b02e7f65c5bb"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-16T11:40:09.925307"
    }
}